pkg/auth/dex_test.go

```diff
@@ -0,0 +1,135 @@
+package auth
+
+import (
+	"bytes"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+func TestNewDexAppClient(t *testing.T) {
+	DEX_URL, _ := url.Parse(dexServiceURL)
+	testCases := []struct {
+		Name          string
+		clientID      string
+		clientSecret  string
+		baseURL       string
+		scopes        []string
+		wantErr       bool
+		wantClientApp *DexAppClient
+	}{
+		{
+			Name:         "Creates the a new Dex App Client as expected",
+			clientID:     "test-client",
+			clientSecret: "test-secret",
+			baseURL:      "www.test-url.com",
+			scopes:       []string{"scope1", "scope2"},
+			wantErr:      false,
+			wantClientApp: &DexAppClient{
+				ClientID:     "test-client",
+				ClientSecret: "test-secret",
+				RedirectURI:  "www.test-url.com/callback",
+				IssuerURL:    "www.test-url.com/dex",
+				BaseURL:      "www.test-url.com",
+				Scopes:       []string{"scope1", "scope2"},
+				Client: &http.Client{
+					Transport: DexRewriteURLRoundTripper{
+						DexURL: DEX_URL,
+						T:      http.DefaultTransport,
+					},
+				},
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			a, err := NewDexAppClient(tc.clientID, tc.clientSecret, tc.baseURL, tc.scopes)
+			if (err != nil) != tc.wantErr {
+				t.Errorf("build dependency map error = %v, wantErr %v", err, tc.wantErr)
+			}
+			if diff := cmp.Diff(a, tc.wantClientApp, cmpopts.IgnoreFields(DexRewriteURLRoundTripper{}, "T")); diff != "" {
+				t.Errorf("got %v, want %v, diff (-want +got) %s", a, tc.wantClientApp, diff)
+			}
+		})
+	}
+}
+
+func TestNewDexReverseProxy(t *testing.T) {
+	testCases := []struct {
+		Name           string
+		mockDexServer  *httptest.Server
+		wantErr        bool
+		wantStatusCode int
+	}{
+		{
+			Name: "Dex reverse proxy is working as expected on success",
+			mockDexServer: httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+				rw.WriteHeader(http.StatusOK)
+			})),
+			wantStatusCode: http.StatusOK,
+			wantErr:        false,
+		},
+		{
+			Name: "Dex reverse proxy is working as expected on error",
+			mockDexServer: httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+				rw.WriteHeader(http.StatusInternalServerError)
+			})),
+			wantStatusCode: http.StatusInternalServerError,
+			wantErr:        false,
```

Comment:
Just a comment, I understand the wantErr differs from the error status code but it can be misleading. I would be expecting that when the status code is 500 that I want an error in fact. I may be nitpicking though.

Reply:
Yes, you are right. The `wantErr` was not doing anything in there. I removed the variable from the test.